lcof2/剑指 Offer II 057. 值和下标之差都在给定的范围内/README.md

```diff
@@ -809,6 +809,53 @@ class TreeMultiSet<T = number> {
 }
 ```
 
+#### Swift
+
+```swift
+class Solution {
+    func containsNearbyAlmostDuplicate(_ nums: [Int], _ k: Int, _ t: Int) -> Bool {
+        guard nums.count > 1, k > 0, t >= 0 else { return false }
+
+        var ts = SortedSet<Int64>()
+        for i in 0..<nums.count {
+            let num = Int64(nums[i])
+            if let x = ts.ceiling(num - Int64(t)), x <= num + Int64(t) {
+                return true
+            }
+            ts.insert(num)
+            if i >= k {
+                ts.remove(Int64(nums[i - k]))
+            }
+        }
+        return false
+    }
+}
+
+struct SortedSet<Element: Comparable> {
+    private var elements = [Element]()
+    
+    mutating func insert(_ value: Element) {
+        elements.append(value)
+        elements.sort()
+    }
+    
+    mutating func remove(_ value: Element) {
+        if let index = elements.firstIndex(of: value) {
+            elements.remove(at: index)
+        }
+    }
+    
+    func ceiling(_ value: Element) -> Element? {
+        for element in elements {
+            if element >= value {
+                return element
+            }
+        }
+        return nil
+    }
+}
+```
```

Comment:
@klever34 

Hello, the SortedSet implemented in this way has low efficiency and cannot pass all test cases within the time limit. It needs to be reimplemented.

Reply:
https://leetcode.com/problems/contains-duplicate-iii/

Reply:
Thanks. It has been taken care of. Kindly review again.

Reply:
![image](https://github.com/doocs/leetcode/assets/21008209/dc5464e1-c94f-48eb-a976-2d979c3578d6)

:(

Reply:
Interesting, let me take a look at it.